langchain4j-redis/src/main/java/dev/langchain4j/store/embedding/redis/RedisSchema.java

```diff
@@ -64,8 +66,14 @@ SchemaField[] toSchemaFields() {
                 .build());
 
         if (metadataKeys != null) {
+            metadataKeys.forEach(metadataKey -> {
+                if (!schemaFieldMap.containsKey(metadataKey)) {
+                    throw new RuntimeException("Schema field " + metadataKey + " not found");
+                }
+            });
             for (String metadataKey : metadataKeys) {
-                fields.add(TextField.of(JSON_PATH_PREFIX + metadataKey).as(metadataKey).weight(1.0));
```

Comment:
what about keeping the old logic if `metadataKeys` was not provided by the user?

Reply:
Nice point. Already done.